File: VRSampleApp/VRSampleApp/Classes/BusinessLogicLayer/Services/TestdataService/TestDataService.m
//
//  TestDataService.m
//  VRSampleApp
//
//  Copyright © 2017 Ooyala Inc. All rights reserved.
//

#import "TestDataService.h"
#import "VideoItemSection+Mappable.h"


@implementation TestDataService

#pragma mark - TestDataServiceProtocol

- (NSArray *)obtainTestData {
  NSString *pathToTestDataJSON = [NSBundle.mainBundle pathForResource:@"data" ofType:@"json"];
  
  if (!pathToTestDataJSON) {
    return NULL;
  }
  
  NSURL *urlToTestDataJSON = [NSURL fileURLWithPath:pathToTestDataJSON];
  NSData *jsonData = [NSData dataWithContentsOfURL:urlToTestDataJSON];
  
  if (!jsonData) {
    return NULL;
  }
  
  NSError *error;
  NSDictionary *json = [NSJSONSerialization JSONObjectWithData:jsonData
                                                       options:NSJSONReadingAllowFragments
                                                         error:&error];
  
  return [VideoItemSection createCollectionFromJSON:json];
}


@end

File: SwiftVRSkinSampleApp/SwiftVRSkinSampleApp/Classes/BusinessLogicLayer/Services/TestDataService/TestDataService.swift
//
//  TestDataService.swift
//  VRSkinSampleApp
//
//  Copyright © 2017 Ooyala, Inc. All rights reserved.
//

import Foundation

class TestDataService {
}

// MARK: - TestDataServiceProtocol
extension TestDataService: TestDataServiceProtocol {

  var testData: [VideItemSection] {
    guard let pathToTestDataJSON = Bundle.main.path(forResource: "data",
                                                    ofType: "json") else { return [] }

    do {
      let urlToTestDataJSON = URL(fileURLWithPath: pathToTestDataJSON)
      let jsonData = try Data(contentsOf: urlToTestDataJSON)
      let json = try JSONSerialization.jsonObject(with: jsonData,
                                                  options: .allowFragments)

      return VideItemSection.createCollection(fromJSON: json) ?? []
    } catch {
      return []
    }
  }
}
